test/core/container.jl

```diff
@@ -10,76 +10,74 @@ include(dir*"/test/test_utils/AllUtils.jl")
 
 @testset "container.jl" begin
     @turing_testset "copy particle container" begin
-        pc = ParticleContainer(x -> x * x, Trace[])
+        pc = ParticleContainer(Trace[])
         newpc = copy(pc)
 
         @test newpc.logWs == pc.logWs
         @test typeof(pc) === typeof(newpc)
     end
     @turing_testset "particle container" begin
-        n = Ref(0)
-
-        alg = PG(5)
-        spl = Turing.Sampler(alg, empty_model())
-        dist = Normal(0, 1)
-
-        function fpc()
-            t = TArray(Float64, 1);
-            t[1] = 0;
-            while true
-                ct = current_trace()
-                vn = @varname x[n]
-                Turing.assume(spl, dist, vn, ct.vi); n[] += 1;
-                produce(0)
-                vn = @varname x[n]
-                Turing.assume(spl, dist, vn, ct.vi); n[] += 1;
-                t[1] = 1 + t[1]
+        # Create a resumable function that always yields `logp`.
+        function fpc(logp)
+            f = let logp = logp
+                () -> begin
+                    while true
+                        produce(logp)
+                    end
+                end
             end
+            return f
         end
 
-        modelgen = Turing.ModelGen{()}(nothing, NamedTuple())
-        model = Turing.Model(fpc, NamedTuple(), modelgen)
-        particles = [Trace(fpc, model, spl, Turing.VarInfo()) for _ in 1:3]
-        pc = ParticleContainer(fpc, particles)
+        # Dummy sampler that is not actually used.
+        sampler = Sampler(PG(5), empty_model())
 
-        # Initial weights and likelihood.
-        weights = getweights(pc)
-        lz = logZ(pc)
-        @test weights == [1/3, 1/3, 1/3]
-        @test iszero(lz)
+        # Create particle container.
+        logps = [0.0, -1.0, -2.0]
+        particles = [Trace(fpc(logp), empty_model(), sampler, VarInfo()) for logp in logps]
```

Comment:
This looks good. On a side note, it's strange that we still have the following two `Trace` constructors:

```julia
function Trace(f::Function, m::Model, spl::AbstractSampler, vi::AbstractVarInfo)
function Trace(m::Model, spl::AbstractSampler, vi::AbstractVarInfo)
```

maybe we can consider merging these two in the future.

Reply:
Yep, that's a bit strange indeed.